src/components/Attachment/Audio.tsx

```diff
@@ -1,13 +1,22 @@
-// @ts-check
 import React, { useCallback, useEffect, useRef, useState } from 'react';
 
+import type { Attachment } from 'stream-chat';
+
+import type { DefaultAttachmentType, UnknownType } from '../../../types/types';
+
 const progressUpdateInterval = 500;
 /**
  * Audio attachment with play/pause button and progress bar
- * @param {import("types").AudioProps} props
  */
-const Audio = ({ og }) => {
-  const audioRef = useRef(/** @type {HTMLAudioElement | null} */ (null));
+export type AudioProps<At extends UnknownType = DefaultAttachmentType> = {
+  og: Attachment<At>;
```

Comment:
looks like Audio is expecting the default generic, but the ExtendedAttachment type in Attachment isn't generic, to be safe and cover bases should prob enable ExtendedAttachment to take At otherwise we're always going to be using the default and not a user's custom type

Reply:
Alright, updated and I think I got everything threaded through correctly there.